libs/SmartStore/SmartStore/Classes/SFStoreCursor.m

```diff
@@ -42,112 +39,57 @@ @interface SFStoreCursor ()
         [self close];
 }
 
 
 - (void)close {
     [SFSDKSmartStoreLogger v:[self class] format:@"closing cursor id: %@",self.cursorId];
-     _store = nil;
     self.cursorId = nil;
     self.querySpec = nil;
-    self.currentPageOrderedEntries = nil;
     self.currentPageIndex = nil;
     self.pageSize = nil;
     self.totalPages = nil;
 }
 
-#pragma mark - Properties
-
-- (void)setCurrentPageIndex:(NSNumber *)pageIdx {
-    //TODO check bounds?
-    if (![pageIdx isEqual:_currentPageIndex]) {
-        _currentPageIndex = pageIdx;
-        
-        if (nil != _currentPageIndex) {
-            if ([self.totalPages integerValue] > 0) {
-                NSUInteger pageIdx = [_currentPageIndex integerValue];
-                NSArray *newEntries = [_store queryWithQuerySpec:self.querySpec pageIndex:pageIdx error:nil];
-                self.currentPageOrderedEntries = newEntries;
-            } else {
-                self.currentPageOrderedEntries = @[];
-            }
-        } 
-    }
-}
-
-#pragma mark - Converting to JSON
-
-- (NSDictionary*)asDictionary
-{
-    NSMutableDictionary *result = [NSMutableDictionary dictionary];
-    result[@"cursorId"] = self.cursorId;
-    //note that we only encode the current page worth of entries
-    result[@"currentPageOrderedEntries"] = self.currentPageOrderedEntries ?: @[];
-    result[@"currentPageIndex"] = self.currentPageIndex ?: @0;
-    result[@"pageSize"] = self.pageSize ?: @0;
-    result[@"totalPages"] = self.totalPages ?: @0;
-    result[@"totalEntries"] = self.totalEntries ?: @0;
-    
-    return result;
-}
-
-
-- (NSString*)description {
-    return [NSString stringWithFormat:@"<SFStoreCursor: %p> {\n cursorId: %@ \n totalEntries:%@ \n totalPages:%@ \n currentPage:%@ \n currentPageOrderedEntries: [%lu] \n querySpec: %@ \n }",
-            self,self.cursorId,
-            self.totalEntries,
-            self.totalPages,
-            self.currentPageIndex,
-            (unsigned long)[self.currentPageOrderedEntries count],
-            self.querySpec
-            ];
+- (NSString*)getDataSerialized:(SFSmartStore*)store error:(NSError**)error {
+    NSMutableString* resultBuilder = [NSMutableString new];
+    [resultBuilder appendString:@"{"];
+    [resultBuilder appendFormat:@"\"%@\":\"%@\", ", @"cursorId", self.cursorId];
+    [resultBuilder appendFormat:@"\"%@\":%@, ", @"currentPageIndex", self.currentPageIndex ?: @0];
+    [resultBuilder appendFormat:@"\"%@\":%@, ", @"pageSize", self.pageSize ?: @0];
+    [resultBuilder appendFormat:@"\"%@\":%@, ", @"totalPages", self.totalPages ?: @0];
+    [resultBuilder appendFormat:@"\"%@\":%@, ", @"totalEntries", self.totalEntries ?: @0];
+    [resultBuilder appendFormat:@"\"%@\":", @"currentPageOrderedEntries"];
+    [store queryAsString:resultBuilder querySpec:self.querySpec pageIndex:[self.currentPageIndex integerValue] error:error];
+    [resultBuilder appendString:@"}"];
+    return resultBuilder;
```

Comment:
@wmathurin do you think I should add JSON validation check here as well ?

Reply:
I think it would be overkill if we already checked that the query-as-string method returned something that is a properly json-ized object.